packages/sanity/src/core/form/inputs/files/common/UploadPlaceholder.tsx

```diff
@@ -1,63 +1,76 @@
-import React from 'react'
+import React, {useState} from 'react'
 import {UploadIcon} from '@sanity/icons'
-import {Flex, Inline} from '@sanity/ui'
+import {Flex, useElementSize} from '@sanity/ui'
 import {FileLike} from '../../../studio/uploads/types'
 import {FileInputButton} from './FileInputButton/FileInputButton'
 import {PlaceholderText} from './PlaceholderText'
 
-type UploadPlaceholderProps = {
-  onUpload?: (files: File[]) => void
+interface UploadPlaceholderProps {
+  accept: string
+  acceptedFiles: FileLike[]
   browse?: React.ReactNode
-  readOnly?: boolean
-  type: string
+  directUploads?: boolean
   hoveringFiles: FileLike[]
-  acceptedFiles: FileLike[]
+  onUpload?: (files: File[]) => void
+  readOnly?: boolean
   rejectedFilesCount: number
-  accept: string
-  directUploads?: boolean
+  type: string
 }
 
-export const UploadPlaceholder = React.memo(function UploadPlaceholder({
-  browse,
-  onUpload,
-  readOnly,
-  type,
-  hoveringFiles,
-  acceptedFiles,
-  rejectedFilesCount,
-  accept,
-  directUploads,
-}: UploadPlaceholderProps) {
+function UploadPlaceholderComponent(props: UploadPlaceholderProps) {
+  const {
+    accept,
+    acceptedFiles,
+    browse,
+    directUploads,
+    hoveringFiles,
+    onUpload,
+    readOnly,
+    rejectedFilesCount,
+    type,
+  } = props
+
+  const [rootElement, setRootElement] = useState<HTMLDivElement | null>(null)
+  const rect = useElementSize(rootElement)
+
+  // Adjust the layout in narrow containers
+  const collapsed = rect?.border && rect.border.width < 440
```

Comment:
I think there are specific values provided by the theme / sanity ui that we can use instead of a flat value
`sanity.media` or something along those lines? 🤔

Reply:
Yes, good point 👍   

However, in this situation, we need a specific value that fits the design of the image and file input. We want to change the layout when the component starts to get squeezed, and no value in `sanity.media` works well to accomplish that :)

Reply:
Ah I see :) in that case disregard this comment ✨️